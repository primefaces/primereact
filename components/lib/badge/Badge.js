import React, { forwardRef, memo } from 'react';
import PropTypes from 'prop-types';
import { classNames } from '../utils/Utils';

export const Badge = memo(forwardRef((props, ref) => {
    const className = classNames('p-badge p-component', {
        'p-badge-no-gutter': props.value && String(props.value).length === 1,
        'p-badge-dot': !props.value,
        'p-badge-lg': props.size === 'large',
        'p-badge-xl': props.size === 'xlarge',
        [`p-badge-${props.severity}`]: props.severity !== null
    }, props.className);

    return (
        <span className={className} style={props.style}>
            {props.value}
        </span>
    )
}));

Badge.defaultProps = {
    __TYPE: 'Badge',
    value: null,
    severity: null,
    size: null,
    style: null,
    className: null
}

Badge.propTypes /* remove-proptypes */ = {
    __TYPE: PropTypes.string,
    value: PropTypes.any,
    severity: PropTypes.string,
    size: PropTypes.string,
    style: PropTypes.object,
    className: PropTypes.string
}
