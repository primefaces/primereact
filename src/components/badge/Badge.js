import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { classNames } from '../utils/Utils';

export class Badge extends Component {

    static defaultProps = {
        value: null,
        severity: null,
        size: null,
        style: null,
        className: null
    }

    static propTypes = {
        value: PropTypes.any,
        severity: PropTypes.string,
        size: PropTypes.string,
        style: PropTypes.object,
        className: PropTypes.string
    };

    render() {
        const badgeClassName = classNames('p-badge p-component', {
            'p-badge-no-gutter': this.props.value && String(this.props.value).length === 1,
            'p-badge-dot': !this.props.value,
            'p-badge-lg': this.props.size === 'large',
            'p-badge-xl': this.props.size === 'xlarge',
            'p-badge-info': this.props.severity === 'info',
            'p-badge-success': this.props.severity === 'success',
            'p-badge-warning': this.props.severity === 'warning',
            'p-badge-danger': this.props.severity === 'danger'
        }, this.props.className);

        return (
            <span className={badgeClassName} style={this.props.style}>
                {this.props.value}
            </span>
        );
    }
}
