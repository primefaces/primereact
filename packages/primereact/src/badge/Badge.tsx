'use client';
import { Component } from '@primereact/core/component';
import { useBadge } from '@primereact/headless/badge';
import { styles } from '@primereact/styles/badge';
import { mergeProps } from '@primeuix/utils';
import { withComponent } from 'primereact/base';
import * as React from 'react';
import { BadgeProvider } from './Badge.context';
import { defaultProps } from './Badge.props';
import { OverlayBadge } from './overlay';

export const Badge = withComponent({
    name: 'Badge',
    defaultProps,
    styles,
    setup(instance) {
        const badge = useBadge(instance.inProps);

        return badge;
    },
    render(instance) {
        const { id, props, ptmi, cx } = instance;

        const rootProps = mergeProps(
            {
                id,
                className: cx('root')
            },
            ptmi('root')
        );

        return (
            <BadgeProvider value={instance}>
                <Component instance={instance} attrs={rootProps} children={props.children} />
            </BadgeProvider>
        );
    },
    components: {
        Overlay: OverlayBadge
    }
});
