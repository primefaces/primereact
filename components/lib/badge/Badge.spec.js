import '@testing-library/jest-dom';
import { Badge } from './Badge';

import { snapshot } from '../../test';

describe('Badge', () => {
    snapshot(<Badge />, 'default');
    snapshot(<Badge value={22} />, 'value');
    snapshot(<Badge size="large" />, 'size large');
    snapshot(<Badge size="xlarge" />, 'size xlarge');
    snapshot(<Badge size="invalid" />, 'size invalid');
    snapshot(<Badge severity="success" />, 'severity success');
    snapshot(<Badge severity="info" />, 'severity info');
    snapshot(<Badge severity="warning" />, 'severity warning');
    snapshot(<Badge severity="danger" />, 'severity danger');
    snapshot(<Badge severity="invalid" />, 'severity invalid');
});
