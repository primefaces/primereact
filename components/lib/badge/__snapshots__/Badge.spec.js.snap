// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge default 1`] = `
<div>
  <span
    class="p-badge p-component p-badge-dot"
  />
</div>
`;

exports[`Badge severity danger 1`] = `
<div>
  <span
    class="p-badge p-component p-badge-dot p-badge-danger"
  />
</div>
`;

exports[`Badge severity info 1`] = `
<div>
  <span
    class="p-badge p-component p-badge-dot p-badge-info"
  />
</div>
`;

exports[`Badge severity invalid 1`] = `
<div>
  <span
    class="p-badge p-component p-badge-dot p-badge-invalid"
  />
</div>
`;

exports[`Badge severity success 1`] = `
<div>
  <span
    class="p-badge p-component p-badge-dot p-badge-success"
  />
</div>
`;

exports[`Badge severity warning 1`] = `
<div>
  <span
    class="p-badge p-component p-badge-dot p-badge-warning"
  />
</div>
`;

exports[`Badge size invalid 1`] = `
<div>
  <span
    class="p-badge p-component p-badge-dot"
  />
</div>
`;

exports[`Badge size large 1`] = `
<div>
  <span
    class="p-badge p-component p-badge-dot p-badge-lg"
  />
</div>
`;

exports[`Badge size xlarge 1`] = `
<div>
  <span
    class="p-badge p-component p-badge-dot p-badge-xl"
  />
</div>
`;

exports[`Badge value 1`] = `
<div>
  <span
    class="p-badge p-component"
  >
    22
  </span>
</div>
`;
